<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>search_by_maps</class>
 <widget class="QWidget" name="search_by_maps">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1312</width>
    <height>657</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QTabWidget" name="tab_main">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>200</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_select_map">
       <attribute name="title">
        <string>Select Event</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="MplCanvasMapWithToolbar" name="mpl_select_map" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="DateTimeEdit" name="datetime_start">
            <property name="dateTime">
             <datetime>
              <hour>0</hour>
              <minute>0</minute>
              <second>0</second>
              <year>2023</year>
              <month>2</month>
              <day>6</day>
             </datetime>
            </property>
            <property name="displayFormat">
             <string>dd/MM/yyyy HH:mm:ss</string>
            </property>
            <property name="calendarPopup">
             <bool>true</bool>
            </property>
            <property name="timeSpec">
             <enum>Qt::LocalTime</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="DateTimeEdit" name="datetime_end">
            <property name="dateTime">
             <datetime>
              <hour>23</hour>
              <minute>59</minute>
              <second>59</second>
              <year>2023</year>
              <month>2</month>
              <day>6</day>
             </datetime>
            </property>
            <property name="date">
             <date>
              <year>2023</year>
              <month>2</month>
              <day>6</day>
             </date>
            </property>
            <property name="displayFormat">
             <string>dd/MM/yyyy HH:mm:ss</string>
            </property>
            <property name="calendarPopup">
             <bool>true</bool>
            </property>
            <property name="timeSpec">
             <enum>Qt::LocalTime</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_rectangle">
            <property name="text">
             <string>Create Rectangle</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_showevents">
            <property name="text">
             <string>Show Events</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_selectevent">
            <property name="text">
             <string>Select An Event</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_map_reset">
            <property name="text">
             <string>Reset</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QCheckBox" name="chk_showtable">
            <property name="text">
             <string>Show in Table</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_select_table">
       <attribute name="title">
        <string>Select Event (Table)</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QTableView" name="table_events">
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="spacing">
           <number>0</number>
          </property>
          <item>
           <widget class="QPushButton" name="btn_selectevent_tbl">
            <property name="text">
             <string>Select An Event</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_save_events">
            <property name="text">
             <string>Save Events</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_select_plot">
       <attribute name="title">
        <string>Select Event (Plot)</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="MplCanvasMapWithToolbar" name="mpl_select_map_plot" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QPushButton" name="btn_selectevent_plot">
            <property name="text">
             <string>Select An Event</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_5">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_stations">
       <attribute name="title">
        <string>Select Stations</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="MplCanvasMapWithToolbar" name="mpl_select_map_2" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QPushButton" name="btn_rectangle_2">
            <property name="text">
             <string>Create Rectangle</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_showstations">
            <property name="text">
             <string>Show Stations</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Select Stations:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_selectstations">
            <property name="text">
             <string>Free Shape</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_selectstations_circle">
            <property name="text">
             <string>Circle</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="txt_circle">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string/>
            </property>
            <property name="text">
             <string>15</string>
            </property>
            <property name="placeholderText">
             <string>radius (deg)</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QPushButton" name="btn_getwaves">
            <property name="text">
             <string>Get Waveforms</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_showwaves">
            <property name="text">
             <string>Show Waveforms</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_save_stations">
            <property name="text">
             <string>Save Selected Stations</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_save_waveforms">
            <property name="text">
             <string>Save Waveforms</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_6">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_show_waves">
       <attribute name="title">
        <string>Show Waves</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="MplCanvasBaseWithToolbar" name="cv_distance" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>400</width>
            <height>400</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MplCanvasBaseWithToolbar</class>
   <extends>QWidget</extends>
   <header>widgets.mplcanvas</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>MplCanvasMapWithToolbar</class>
   <extends>QWidget</extends>
   <header>widgets.mplcanvas</header>
   <container>1</container>
   <slots>
    <slot>select_rectangle(bool)</slot>
    <slot>select_event(bool)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>DateTimeEdit</class>
   <extends>QDateTimeEdit</extends>
   <header>widgets.datetimeedit</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>btn_showwaves</sender>
   <signal>clicked()</signal>
   <receiver>search_by_maps</receiver>
   <slot>_show_waveforms()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>633</x>
     <y>330</y>
    </hint>
    <hint type="destinationlabel">
     <x>640</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_showstations</sender>
   <signal>clicked()</signal>
   <receiver>search_by_maps</receiver>
   <slot>_on_btn_show_stations_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>215</x>
     <y>330</y>
    </hint>
    <hint type="destinationlabel">
     <x>640</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_showevents</sender>
   <signal>clicked()</signal>
   <receiver>search_by_maps</receiver>
   <slot>_on_btn_show_events_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>556</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>640</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_selectstations</sender>
   <signal>toggled(bool)</signal>
   <receiver>search_by_maps</receiver>
   <slot>_select_stations()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>350</x>
     <y>330</y>
    </hint>
    <hint type="destinationlabel">
     <x>640</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_selectevent_plot</sender>
   <signal>toggled(bool)</signal>
   <receiver>mpl_select_map_plot</receiver>
   <slot>select_event(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>71</x>
     <y>330</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>170</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_selectevent</sender>
   <signal>toggled(bool)</signal>
   <receiver>mpl_select_map</receiver>
   <slot>select_event(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>687</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_rectangle_2</sender>
   <signal>toggled(bool)</signal>
   <receiver>mpl_select_map_2</receiver>
   <slot>select_rectangle(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>75</x>
     <y>330</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>170</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_rectangle</sender>
   <signal>toggled(bool)</signal>
   <receiver>mpl_select_map</receiver>
   <slot>select_rectangle(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>420</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_map_reset</sender>
   <signal>clicked()</signal>
   <receiver>search_by_maps</receiver>
   <slot>_on_btn_map_reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>797</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>640</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_getwaves</sender>
   <signal>clicked()</signal>
   <receiver>search_by_maps</receiver>
   <slot>_on_btn_get_waveforms_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>488</x>
     <y>330</y>
    </hint>
    <hint type="destinationlabel">
     <x>640</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_save_stations</sender>
   <signal>clicked()</signal>
   <receiver>search_by_maps</receiver>
   <slot>_save_stations()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>114</x>
     <y>330</y>
    </hint>
    <hint type="destinationlabel">
     <x>643</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_save_waveforms</sender>
   <signal>clicked()</signal>
   <receiver>search_by_maps</receiver>
   <slot>_save_waveforms()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>328</x>
     <y>330</y>
    </hint>
    <hint type="destinationlabel">
     <x>643</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_selectevent_tbl</sender>
   <signal>clicked()</signal>
   <receiver>search_by_maps</receiver>
   <slot>_on_btn_selectevent_tbl_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>71</x>
     <y>637</y>
    </hint>
    <hint type="destinationlabel">
     <x>655</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_save_events</sender>
   <signal>clicked()</signal>
   <receiver>search_by_maps</receiver>
   <slot>_on_btn_save_events_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>207</x>
     <y>637</y>
    </hint>
    <hint type="destinationlabel">
     <x>655</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_selectstations_circle</sender>
   <signal>clicked()</signal>
   <receiver>search_by_maps</receiver>
   <slot>_on_btn_selectstations_circle_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>557</x>
     <y>604</y>
    </hint>
    <hint type="destinationlabel">
     <x>655</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>_show_waveforms()</slot>
  <slot>_on_btn_show_stations_clicked()</slot>
  <slot>_on_btn_show_events_clicked()</slot>
  <slot>_select_stations()</slot>
  <slot>run_kernel()</slot>
  <slot>_on_btn_map_reset()</slot>
  <slot>_on_btn_get_waveforms_clicked()</slot>
  <slot>_save_stations()</slot>
  <slot>_save_waveforms()</slot>
  <slot>_on_btn_selectevent_tbl_clicked()</slot>
  <slot>_on_btn_save_events_clicked()</slot>
  <slot>_on_btn_selectstations_circle_clicked()</slot>
 </slots>
</ui>
